src/components/App/index.spec.tsx

```diff
@@ -35,71 +40,125 @@ describe(__filename, () => {
     expect(root.find(`.${styles.content}`)).toHaveLength(1);
     expect(root.find(Navbar)).toHaveLength(1);
   });
 
-  it('displays a loading message until the user profile gets loaded', () => {
-    const root = render();
+  it('displays a loading message until the user profile gets loaded', async () => {
+    const root = await render();
 
     expect(root).toIncludeText('Getting your workspace ready');
     expect(root.find(Navbar)).toHaveLength(0);
   });
 
-  it('dispatches setAuthToken on mount when authToken is valid', () => {
+  it('dispatches setAuthToken on mount when authToken is valid', async () => {
     const authToken = 'my-token';
     const store = configureStore();
     const dispatch = jest.spyOn(store, 'dispatch');
 
-    render({ authToken, store });
+    await render({ authToken, store });
 
     expect(dispatch).toHaveBeenCalledWith(
       apiActions.setAuthToken({ authToken }),
     );
   });
 
-  it('does not dispatch setAuthToken on mount when authToken is null', () => {
+  it('does not dispatch setAuthToken on mount when authToken is null', async () => {
     const store = configureStore();
     const dispatch = jest.spyOn(store, 'dispatch');
 
-    render({ authToken: null, store });
+    await render({ authToken: null, store });
 
     expect(dispatch).not.toHaveBeenCalled();
   });
 
-  it('configures no route when the user is not logged in', () => {
-    const root = render({ authToken: null });
+  it('configures no route when the user is not logged in', async () => {
+    const root = await render({ authToken: null });
 
     expect(root.find(Route)).toHaveLength(0);
     expect(root.find(`.${styles.loginMessage}`)).toIncludeText('Please log in');
   });
 
-  it('exposes more routes when the user is logged in', () => {
+  it('exposes more routes when the user is logged in', async () => {
     const store = configureStore();
     store.dispatch(userActions.loadCurrentUser({ user: fakeUser }));
 
-    const root = render({ store });
+    const root = await render({ store });
 
     expect(root.find(Route)).toHaveLength(3);
     expect(root.find(`.${styles.loginMessage}`)).toHaveLength(0);
   });
+
+  it('dispatches loadCurrentUser() on update', async () => {
+    const _log = getFakeLogger();
+    const authToken = 'my-token';
+    const user = fakeUser;
+
+    const store = configureStore();
+    const dispatch = jest.spyOn(store, 'dispatch');
+
+    const mockApi = jest.spyOn(api, 'getCurrentUserProfile');
+    mockApi.mockReturnValue(Promise.resolve(user));
+
+    const root = await render({ _log, authToken: null, store });
+
+    // Set the authentication token into the state.
+    store.dispatch(apiActions.setAuthToken({ authToken }));
+    const { api: apiState } = store.getState();
+    dispatch.mockReset();
+
+    // Wait until componentDidUpdate() logic has been executed with the new
+    // props.
+    await updateWithProps(root, { apiState });
```

Comment:
I'd rather not go down this route because once we have https://github.com/mozilla/addons-code-manager/issues/12 then we won't need it. Everything can be tested synchronously. However, I guess it's OK as a first take. 

As for side effects, I'm going to suggest using thunks because I don't have a better idea. I will be cleaning up https://github.com/mozilla/addons-code-manager/pull/176 asap.

Reply:
I cleaned up the patch for thunks (https://github.com/mozilla/addons-code-manager/pull/176). Maybe you can build on top of that? If you convert to thunks, I don't think you'll need any async handling in the component or the tests.

Reply:
I can certainly build on top of the thunk patch, I'll do a final review first.